util/network-devp2p/src/connection.rs

```diff
@@ -317,14 +315,13 @@ impl EncryptedConnection {
 		let key_material_keccak = keccak(&key_material);
 		(&mut key_material[32..64]).copy_from_slice(key_material_keccak.as_bytes());
 
-		let iv = vec![0u8; 16];
-		let encoder = CtrMode::new(AesSafe256Encryptor::new(&key_material[32..64]), iv);
-		let iv = vec![0u8; 16];
-		let decoder = CtrMode::new(AesSafe256Encryptor::new(&key_material[32..64]), iv);
-
-		let key_material_keccak = keccak(&key_material);
-		(&mut key_material[32..64]).copy_from_slice(key_material_keccak.as_bytes());
-		let mac_encoder = EcbEncryptor::new(AesSafe256Encryptor::new(&key_material[32..64]), NoPadding);
+		// TODO: clarify this: ecdh::agree creates a **NEW** secret right? And AesCtr256 keeps an internal counter, right?
+		// Using a 0 IV with CTR is fine as long as the same IV is never reused with the same key. This is not the case here.
+		let encoder = AesCtr256::new(&key_material[32..64], &NULL_IV)?;
+		let decoder = AesCtr256::new(&key_material[32..64], &NULL_IV)?;
+        let key_material_keccak = keccak(&key_material);
+        (&mut key_material[32..64]).copy_from_slice(key_material_keccak.as_bytes());
```

Comment:
This last copy seems useless (except to show a repeating pattern with previous lines, which can make sense and then we rely on optimization). Not sure if it is worth removing.

Reply:
Yeah it seems like we could use `key_material_keccak` everywhere we use `key_material[32..64]` below. I'll leave it as it is because I don't understand it fully.